README-FIPS.md

```diff
@@ -77,6 +81,78 @@ If you configured OpenSSL to be installed to a different location, the paths wil
 vary accordingly. In the rare case that you need to install the fipsmodule.cnf
 to non-standard location, you can execute the `openssl fipsinstall` command manually.
 
+Installing the FIPS provider and using it with the latest release
+=================================================================
+
+This normally requires you to download 2 copies of the OpenSSL source code.
+
+Download and build the (OpenSSL 3.0) FIPS provider
+--------------------------------------------------
+
+OpenSSL 3.0 is the only source that is currently FIPS validated.
+If you use any other source currently it is NOT FIPS valid.
+Please refer to the security guide for guidance related to
+the OpenSSL 3.0  FIPS provider.
+
+    $ wget https://www.openssl.org/source/openssl-3.0.0.tar.gz
+    $ tar -xf openssl-3.0.0.tar.gz
+    $ cd openssl-3.0.0
+    $ ./Configure enable-fips
+    $ make
+    $ cd ..
+
+Download and build the latest release of OpenSSL
+------------------------------------------------
+
+We use OpenSSL 3.1 here, (but you could also use the latest 3.0.X)
+
+    $ wget https://www.openssl.org/source/openssl-3.1.0.tar.gz
+    $ tar -xf openssl-3.1.0.tar.gz
+    $ cd openssl-3.1.0
+    $ ./Configure enable-fips
+    $ make -j 4
+
+Use the OpenSSL 3.0 FIPS provider for testing
+---------------------------------------------
+
+We do this by replacing the artifact for the OpenSSL 3.1 FIPS provider.
+Note that the OpenSSL 3.1 FIPS provider has not been validated
+so it must not be used for FIPS purposes.
+
+    $ cp ../openssl-3.0.0/providers/fips.so providers/.
+    $ cp ../openssl-3.0.0/providers/fipsmodule.cnf providers/.
```

Comment:
There should be a note somewhere that doing this across machine isn't FIPS valid.

What happens if you just copy the .so across and run `make test` or `fipsinstall`?

Reply:
Actually the providers/fipsmodule.cnf does NOT contain the install checksum, so copying it across machines should not break FIPS compliance - we should fix that in the security policy!

With 3.1.0 it does not even make any sense why it should break compliance as we run the self-tests always.

Reply:
Really?

```
[fips_sect]
activate = 1
conditional-errors = 1
security-checks = 1
ems_check = 0
drgb_no_trunc_dgst = 0
module-mac = 7D:79:91:58:1C:50:6A:21:67:D7:33:61:C0:ED:B0:37:90:5B:FE:54:D1:B2:73:F0:1D:C4:5F:DE:92:4A:15:22
```

With 3.1 we can copy the file if the checks are set to run on load.  We're waiting on the lab to provide us a draft to review and we can make confirm that it's gone.  It should be, I don't remember it in the draft we do have.

Reply:
This still needs addressing.

Reply:
It is there in 3.0 however. and the instructions need to work there.  In fact they are creating this file using 3.0.0 fipsinstall...

Reply:
Is it there in 3.0.0? Did you verify? IMO the `install-mac` is not in the providers/fipsmodule.cnf either because the file is NOT created by fipsinstall but by a separate tool that just computes the module checksum, not the config checksum.